app/main/views/suppliers.py

```diff
@@ -120,17 +121,28 @@ def company_details_from_supplier_framework_declaration(declaration):
             },
         }
 
+    def interesting_frameworks(all_frameworks):
+        if current_user.has_role("admin-ccs-sourcing"):
+            frameworks = (
+                supplier_framework for supplier_framework in all_frameworks
+                if supplier_framework["framework"]["status"] != "coming"
+            )
+        else:
+            frameworks = (
+                supplier_framework for supplier_framework in supplier_frameworks
+                if supplier_framework["framework"]["status"] in ["live", "expired"]
+            )
+        oldest_interesting_id = data_api_client.get_framework(OLDEST_INTERESTING_FRAMEWORK_SLUG)['frameworks']['id']
+        return [framework for framework in frameworks if framework['framework']['id'] >= oldest_interesting_id]
```

Comment:
So many frameworks 😹 But these are `supplier_framework` objects, right? Might be better to rename here.